prometrix/auth.py

```diff
@@ -0,0 +1,76 @@
+import logging
+from typing import Dict
+
+import requests
+
+from prometrix.models.prometheus_config import (AzurePrometheusConfig,
+                                                CoralogixPrometheusConfig,
+                                                PrometheusConfig)
+
+
+class PrometheusAuthorization:
+    bearer_token: str = ""
+
+    @classmethod
+    def azure_authorization(cls, config: PrometheusConfig) -> bool:
+        if not isinstance(config, AzurePrometheusConfig):
+            return False
+        return (config.azure_client_id != "" and config.azure_tenant_id != "") and (
+            config.azure_client_secret != "" or config.azure_use_managed_id != ""
+        )
+
+    @classmethod
+    def get_authorization_headers(cls, config: PrometheusConfig) -> Dict:
+        if isinstance(config, CoralogixPrometheusConfig):
+            return {"token": config.prometheus_token}
+        elif config.prometheus_auth:
+            return {"Authorization": config.prometheus_auth.get_secret_value()}
+        elif cls.azure_authorization(config):
+            return {"Authorization": (f"Bearer {cls.bearer_token}")}
+        else:
+            return {}
+
+    @classmethod
+    def request_new_token(cls, config: PrometheusConfig) -> bool:
+        if cls.azure_authorization(config) and isinstance(
+            config, AzurePrometheusConfig
+        ):
+            try:
+                if config.azure_use_managed_id:
+                    res = requests.get(
```

Comment:
This code is too indented - consider moving `get` and `post` to private methods

Reply:
fixed